packages/typespec-test/test/openai_modular/generated/typespec-ts/review/openai_modular.api.md

```diff
@@ -39,74 +164,280 @@ export interface AzureCognitiveSearchIndexFieldMappingOptions {
+    promptFilterResults?: ContentFilterResultsForPrompt[];
+    systemFingerprint?: string;
     usage: CompletionsUsage;
 }
 
+// @public
+export interface ChatCompletionsFunctionToolCall extends ChatCompletionsToolCall {
+    function: FunctionCall;
+    type: "function";
+}
+
+// @public
+export interface ChatCompletionsFunctionToolDefinition extends ChatCompletionsToolDefinition {
+    function: FunctionDefinition;
+    type: "function";
+}
+
+// @public
+export interface ChatCompletionsJsonResponseFormat extends ChatCompletionsResponseFormat {
+    type: "json_object";
+}
+
+// @public
+export interface ChatCompletionsNamedFunctionToolSelection extends ChatCompletionsNamedToolSelection {
+    name: string;
+    type: "function";
+}
+
+// @public
+export interface ChatCompletionsNamedToolSelection {
+    type: string;
+}
+
+// @public
+export type ChatCompletionsNamedToolSelectionUnion = ChatCompletionsNamedFunctionToolSelection | ChatCompletionsNamedToolSelection;
+
 // @public
 export interface ChatCompletionsOptions {
-    dataSources?: AzureChatExtensionConfiguration[];
+    dataSources?: AzureChatExtensionConfigurationUnion[];
+    enhancements?: AzureChatEnhancementConfiguration;
     frequencyPenalty?: number;
     functionCall?: FunctionCallPreset | FunctionName;
     functions?: FunctionDefinition[];
     logitBias?: Record<string, number>;
     maxTokens?: number;
-    messages: ChatMessage[];
+    messages: ChatRequestMessageUnion[];
     model?: string;
     n?: number;
     presencePenalty?: number;
+    responseFormat?: ChatCompletionsResponseFormatUnion;
+    seed?: number;
     stop?: string[];
     stream?: boolean;
     temperature?: number;
+    toolChoice?: ChatCompletionsToolSelectionPreset | ChatCompletionsNamedToolSelectionUnion;
+    tools?: ChatCompletionsToolDefinitionUnion[];
     topP?: number;
     user?: string;
 }
 
 // @public
-export interface ChatMessage {
+export interface ChatCompletionsResponseFormat {
+    type: string;
+}
+
+// @public
+export type ChatCompletionsResponseFormatUnion = ChatCompletionsTextResponseFormat | ChatCompletionsJsonResponseFormat | ChatCompletionsResponseFormat;
+
+// @public
+export interface ChatCompletionsTextResponseFormat extends ChatCompletionsResponseFormat {
+    type: "text";
+}
+
+// @public
+export interface ChatCompletionsToolCall {
+    id: string;
+    type: string;
+}
+
+// @public
+export type ChatCompletionsToolCallUnion = ChatCompletionsFunctionToolCall | ChatCompletionsToolCall;
+
+// @public
+export interface ChatCompletionsToolDefinition {
+    type: string;
+}
+
+// @public
+export type ChatCompletionsToolDefinitionUnion = ChatCompletionsFunctionToolDefinition | ChatCompletionsToolDefinition;
+
+// @public
+export type ChatCompletionsToolSelectionPreset = string;
+
+// @public
+export interface ChatFinishDetails {
+    type: string;
+}
+
+// @public
+export type ChatFinishDetailsUnion = StopFinishDetails | MaxTokensFinishDetails | ChatFinishDetails;
```

Comment:
Why do we have to include ChatFinishDetails as part of the union? IMO it should only be extended by StopFinishDetails and MaxTokensFinishDetails but not be part of the union